geojson_pydantic/features.py

```diff
@@ -1,25 +1,26 @@
 """pydantic models for GeoJSON Feature objects."""
 
 import json
-from typing import Dict, Generic, List, Optional, TypeVar, Union
+from typing import Any, Dict, Generic, Iterator, List, Optional, Type, TypeVar, Union
 
-from pydantic import Field, ValidationError, validator
+from pydantic import BaseModel, Field, ValidationError, validator
 from pydantic.generics import GenericModel
 
 from geojson_pydantic.geometries import GeoInterfaceMixin, Geometry, GeometryCollection
 from geojson_pydantic.types import BBox
 
-Props = TypeVar("Props", bound=Dict)
-Geom = TypeVar("Geom", bound=Optional[Union[Geometry, GeometryCollection]])
+Props = TypeVar("Props", bound=Union[Dict[str, Any], BaseModel])
+Geom = TypeVar("Geom", bound=Union[Geometry, GeometryCollection])
+F = TypeVar("F", bound="Feature")
 
 
 class Feature(GenericModel, Generic[Geom, Props], GeoInterfaceMixin):
     """Feature Model"""
 
-    type: str = Field("Feature", const=True)
-    geometry: Geom = None
-    properties: Optional[Props]
-    id: Optional[str]
+    type: str = Field(default="Feature", const=True)
+    geometry: Optional[Geom] = None
+    properties: Optional[Props] = None
+    id: Optional[str] = None
```

Comment:
kinda the opposite, if Optional, those default to `None` so maybe we don't need to write it 🤷

Reply:
Yeah, it's a little awkward. From a typing perspective the `Optional` let's it know it can be a `None` value but if you don't have a default value of `= None` they still treats it as a required argument when using the class. Pydantic doesn't care either way but they do.